File: comguanmuonehours/src/main/java/com/guanmu/onehours/activity/SetMissionActivity.java
package com.guanmu.onehours.activity;

import android.os.Bundle;
import androidx.appcompat.widget.Toolbar;
import android.view.View;
import android.view.View.OnClickListener;
import android.widget.SeekBar;
import android.widget.TextView;
import android.widget.Toast;

import com.guanmu.onehours.R;
import com.guanmu.onehours.SharePreferenceUtils;

public class SetMissionActivity extends BaseActivity implements SeekBar.OnSeekBarChangeListener {

    private final int minProgress = 20;
    private final int maxProgress = 200;
    private int num;
    private TextView et_num;
    private SeekBar seekbar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.set_mission_activity);
        seekbar = (SeekBar) findViewById(R.id.seekbar);
        assert seekbar != null;
        seekbar.setMax(maxProgress - minProgress);
        seekbar.setOnSeekBarChangeListener(this);
        num = SharePreferenceUtils.getInstance().getTargetCount();
        seekbar.setProgress(num - minProgress);

        et_num = (TextView) findViewById(R.id.tv_num);
        et_num.setText(String.valueOf(num));

        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        assert toolbar != null;
        toolbar.setNavigationOnClickListener(new OnClickListener() {
            @Override
            public void onClick(View v) {
                finish();
            }
        });
    }


    @Override
    public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
        num = progress + minProgress;
        if (et_num != null)
            et_num.setText(String.valueOf(num));
    }

    @Override
    public void onStartTrackingTouch(SeekBar seekBar) {

    }

    @Override
    public void onStopTrackingTouch(SeekBar seekBar) {
    }


    @Override
    protected void onDestroy() {
        SharePreferenceUtils.getInstance().setTargetCount(num);
        Toast.makeText(this, R.string.tip_setting_saved, Toast.LENGTH_SHORT).show();
        finish();
        super.onDestroy();
    }


}

File: comguanmuonehours/src/main/java/com/guanmu/onehours/activity/SetMissionActivity.kt
package com.guanmu.onehours.activity

import android.os.Bundle
import android.view.View
import android.widget.SeekBar
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.widget.Toolbar
import com.guanmu.onehours.R
import com.guanmu.onehours.SharePreferenceUtils

class SetMissionActivity : BaseActivity(), SeekBar.OnSeekBarChangeListener {

    private val minProgress = 20
    private val maxProgress = 200
    private var num: Int = 0
    private var et_num: TextView? = null
    private var seekbar: SeekBar? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.set_mission_activity)
        seekbar = findViewById<View>(R.id.seekbar) as SeekBar
        assert(seekbar != null)
        seekbar!!.max = maxProgress - minProgress
        seekbar!!.setOnSeekBarChangeListener(this)
        num = SharePreferenceUtils.instance.targetCount
        seekbar!!.progress = num - minProgress

        et_num = findViewById<View>(R.id.tv_num) as TextView
        et_num!!.text = num.toString()

        val toolbar = findViewById<View>(R.id.toolbar) as Toolbar
        setSupportActionBar(toolbar)
        supportActionBar!!.setDisplayHomeAsUpEnabled(true)
        toolbar.setNavigationOnClickListener { finish() }
    }


    override fun onProgressChanged(seekBar: SeekBar, progress: Int, fromUser: Boolean) {
        num = progress + minProgress
        if (et_num != null)
            et_num!!.text = num.toString()
    }

    override fun onStartTrackingTouch(seekBar: SeekBar) {

    }

    override fun onStopTrackingTouch(seekBar: SeekBar) {}


    override fun onDestroy() {
        SharePreferenceUtils.instance.targetCount = num
        Toast.makeText(this, R.string.tip_setting_saved, Toast.LENGTH_SHORT).show()
        finish()
        super.onDestroy()
    }


}
